### PR TITLE
Generate HTML coverage reports

### LINKED ISSUE #56: Add HTML coverage report generation
- [ ] create a variable to control the HTML directory
- [ ] delete the directory when coverage results are cleared
- [ ] generate HTML reports before displaying text reports

<bountysource-plugin>

---
Want to back this issue? **[Place a bounty on it!](https://www.bountysource.com/issues/6637332-add-html-coverage-report-generation?utm_campaign=plugin&utm_content=tracker%2F1487757&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1487757&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by jacebrowning**
Fixes #56

### CODE REVIEW

**Comment by jacebrowning** [comment]
@joshfriend What do you think of this? I know it's not what you suggested, but this seems to work for me and doesn't require another dependency.

File: {{cookiecutter.project_name}}/Makefile [hunk 1]
```diff
@@ -171,26 +171,40 @@ test: test-$(TEST_RUNNER)
...
+
+.PHONY: read-coverage
+read-coverage:
+	$(OPEN) $(COVERAGE_HTML_DIR)/index.html
+
 # nosetest commands
 
 .PHONY: test-nose
 test-nose: depends-ci
+	$(COVERAGE) erase; rm -rf $(COVERAGE_HTML_DIR)
```
Comment (jacebrowning) [inline {{cookiecutter.project_name}}/Makefile]: Or... this target could just depend on `.clean-tests`.

File: {{cookiecutter.project_name}}/Makefile [hunk 2]
```diff
@@ -171,26 +171,38 @@ test: test-$(TEST_RUNNER)
...
 
+.PHONY: read-coverage
+read-coverage:
+	$(OPEN) .coverage-html/index.html
+
 # nosetest commands
 
 .PHONY: test-nose
 test-nose: depends-ci
+	$(COVERAGE) erase; rm -rf .coverage-html
```
Comment (joshfriend) [inline {{cookiecutter.project_name}}/Makefile]: Thoughts on having a `.clean-cov` target that takes care of this and using it as a dependency of these 4 tasks that erase coverage before running?

Comment (joshfriend) [inline {{cookiecutter.project_name}}/Makefile]: or actually, just use `.clean-test`...

**Comment by joshfriend** [comment]
This is basically exactly what I've been doing up until recently when i started using `pytest-cov`. Has worked great for me in the past.

File: {{cookiecutter.project_name}}/Makefile (hunk 2 above)
Comment (jacebrowning) [inline {{cookiecutter.project_name}}/Makefile]: What does `coverage erase` do? Is deleting the `.coverage` equivalent?

Comment (joshfriend) [inline {{cookiecutter.project_name}}/Makefile]: [As far as I can tell, yes.](http://nedbatchelder.com/code/coverage/cmd.html#h_data_file)

**Comment by joshfriend** [comment]
:+1:

**Pull request merged**